basic-services/README.md

```diff
@@ -0,0 +1,176 @@
+
+The intent is to provide an API that is very similar to the
+original MSSDA (Metasys System Secure Data Access) library.
+
+Package name and even org name can still be changed at this time. Suggestions welcome.
+
+
+Installation
+------
+
+Not implemented yet. This is an illustration of desired installation.
+
+
+For dotnet app you can install and use like this:
+
+```bash
+$ dotnet new console
+$ nuget install metasys-basic-services
+```
+
+For Excel App (or any other COM app)
+
+We need a way to install the dll(s) and then also register
+them with COM system.
+
+```bash
+$ nuget install metasys-basic-services
+./register
+```
+
+This approach needs some investigation
+
+1. I assume nuget packages can just be installed "anywhere"? Typically they would be added to some existing
+   dotnet project. But in this case we just need the dll's installed somewhere on box
+2. Need some direction on where they should be installed. Some guidance for the Excel app developer as to what will happen
+3. Guidance on switching to the correct path and then running the ./register script
+4. Guidance on how to add a reference to the dll from excel
+
+Usage
+-------
+
+The following is just some psuedocode. Note, we'll need a resource manager and resource files (work already begun) 
+to provide translations for enums (even for English). 
+
+```csharp
+// Login
+var client = new TraditionalClient();
+
+// apiversion will always have a default based on the release of basic-services, like this initial version will 
+// default to v2. The next major version (released with Metasys 11.0 will likely default to v3)
+// If you need to access both versions then instantiate two clients.
+client.Login(username, password, hostname, apiversion) 
+
+// Read an analog value
+var result = client.ReadProperty(guid, "presentValue")
+Console.Log(result.NumericValue)
+
+// Read an enum value
+var result = client.ReadProperty(guid2, "presentValue")
+var enumValue = result.StringValue
+Console.Log(resourceManger.GetString(enumValue, "en-US")
+
+// Read a string value
+var result = client.ReadProperty(guid3, "description")
+Console.Log(result.StringValue)
+
+// Read a boolean value
+var result = client.ReadProperty(guid4, "isEnabled")
+Console.Log(result.StringValue) // prints "True" or "False" 
+Console.Log(result.NumericValue) // prints 1 or 0
+Console.Log(result.BooleanValue) // prints "True" or "False". This BooleanValue isn't in scope but perhaps it should be
+
+// Read an array value, new for basic services, didn't exist in MSSDA
+var result = client.ReadProperty(guid5, "ipAddress") // ipAddress is an array of bytes
+var firstByteResult = result.ArrayValue
+Console.Log(firstByteResult.Item1)   // prints "192" as Item1 is the string representation
+
+// alternatively
+var (string StringValue, double NumericValue) result = client.ReadProperty(guid5, "ipAddress") // ipAddress is an array of bytes
+Console.Log(result.StringValue)
+
+```
+
+
+Requirements
```

Comment:
There might be something to this. Apparently you can have your libraries marked as trusted in Nuget. Here's a tutorial: https://natemcmaster.com/blog/2018/07/02/code-signing/

So it might be worth doing that.

In general, I've heard that if you want to sign assemblies (to meet some corporate policy) and you are doing open source, then best practice is to just check in your keys. Which wouldn't meet your original intent. (Since anyone could sign anything with our keys).

But the nuget approach seems like maybe we should research just so they know it came from this project and not someone else. (IN either case, I don't think we can sign it using JCI keys)

Reply:
I like the nuget approach you suggested.  It gives a direct consumer of that component some validation of where it originated from.